doc/source/data/glossary.rst

```diff
@@ -0,0 +1,154 @@
+.. _datasets_glossary:
+
+=====================
+Ray Datasets Glossary
+=====================
+
+.. glossary::
+
+    Batch format
+        The way batches of data are represented.
+
+        Set ``batch_format`` in methods like :meth:`~ray.data.Dataset.iter_batches` and
+        :meth:`~ray.data.Dataset.map_batches` to specify the batch type.
```

Comment:
These docs links are a bit ambiguous, it comes out as just `iter_batches()` and `map_batches()`, could we instead make these `Dataset.iter_batches()`? I think that dropping the `~` will still include the module path, so might have to fully qualify.
```suggestion
        Set ``batch_format`` in methods like :meth:`Dataset.iter_batches() <ray.data.Dataset.iter_batches>` and
        :meth:`Dataset.map_batches() <ray.data.Dataset.map_batches>` to specify the batch type.
```